sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2164,10 +2164,19 @@ class Analyzer(
     }
 
     private def hasNestedGenerator(expr: NamedExpression): Boolean = {
+      def hasUnsupportedInnerGenerator(g: Generator): Boolean = g match {
+        case go: GeneratorOuter => // The only case that can have an inner generator
```

Comment:
seems `GeneratorOuter` is just a wrapper not a real `Generator`, so it doesn't count when checking nested generators. Shall we make the comment more clear?

Reply:
yea, ok.